### PR TITLE
feat(clis/transition_tool): Pass `state_test` to transition tool if supported

### PR DESCRIPTION
**Pull request by marioevz**
## 🗒️ Description
Pass the `state_test` property to the t8n that support/require it.

In this PR the only updated tool is EELS, and the information is passed as an URL parameter, which is then converted to a CLI argument on the receiving end.

Requires:
- https://github.com/ethereum/execution-specs/pull/1030
- https://github.com/petertdavies/ethereum-spec-evm-resolver/pull/5

## 🔗 Related Issues
None

## ✅ Checklist
- [x] All: Set appropriate labels for the changes.
- [x] All: Considered squashing commits to improve commit history.
- [x] All: Added an entry to [CHANGELOG.md](/ethereum/execution-spec-tests/blob/main/docs/CHANGELOG.md).
- [x] All: Considered updating the online docs in the [./docs/](/ethereum/execution-spec-tests/blob/main/docs/) directory.
- [x] Tests: All converted JSON/YML tests from [ethereum/tests](/ethereum/tests) have been added to [converted-ethereum-tests.txt](/ethereum/execution-spec-tests/blob/main/converted-ethereum-tests.txt).
- [x] Tests: A PR with removal of converted JSON/YML tests from [ethereum/tests](/ethereum/tests) have been opened.
- [x] Tests: Included the type and version of evm t8n tool used to locally execute test cases:  e.g., ref with commit hash or geth 1.13.1-stable-3f40e65.
- [x] Tests: Ran `mkdocs serve` locally and verified the auto-generated docs for new tests in the [Test Case Reference](https://ethereum.github.io/execution-spec-tests/main/tests/) are correctly formatted.

### CODE REVIEW

**Review by danceratopz — APPROVED** [review]
LTGM. 

But FYI, I didn't try the resolver fork locally.

I did try Besu locally, as I didn't understand why it still worked - from the code it's obvious though, `BesuTransitionTool` doesn't override `_server_post()`, it uses `evaluate() for its implementation. This is a little surprising and the issue to clean this up already exists: #818.

**Comment by marioevz** [comment]
We might need to wait until https://github.com/ethereum/execution-specs/pull/1030 and https://github.com/petertdavies/ethereum-spec-evm-resolver/pull/5 merge before proceeding.

**Comment by marioevz** [comment]
From the passing CI results, I think that we might not have to wait until the other two branches are merged, since we are passing the parameters, but it's not a breaking change in the receiving side since they simply ignored.
Will merge now.

**Pull request merged**
